### DISCUSSION TITLE
Searching a source filter fails when the filter contains two strings.

### DISCUSSION THREAD

**Discussion by Gregory1701** [codefori/vscode-ibmi#2203] [IBM i]
Hey guys,
I really liked that I could setup an object filter to look for objects that start with F. or J. (they have meaning in our shop). So I then thought it would be equally cool to setup a source member filter of the same filter. So I setup the source member filter like this:
![image](https://github.com/user-attachments/assets/df961179-d9b7-4704-8d1e-6984da49bec5)

And it worked! Perfect! But then I wanted to search on QRPGLESRC in that source filter.

This is the resulting error:
![image](https://github.com/user-attachments/assets/fb2d87e8-7542-4803-8ca3-596a0a974fef)

Can we get this working please?

**Comment by sebjulliand** [comment]
I bet we can make it work 😁
First, you have to decide: do you want to use a regular filter (allowing multiple expressions, supporting *), or a regex filter (allowing one regex per field).
You can achieve what you want using one or the other in your case. But in your screenshot, you mixed both approach I think 😊
- Regular filter: F*,J* (everything that starts with F + anything that starts with J)
- Regex filter: (F|J).* or ^(F|J) (both should work: first letter is F or J then anything)

Make sure you select the correct filter type. Let me know how it goes.

**Reply by Gregory1701** [reply]
Hmm, I cannot seem to get this to work with a regular filter or a Regex.
![image](https://github.com/user-attachments/assets/fc0ba7f4-b3aa-457f-b348-01f4aa6ef379)

I tried both of your suggestions for the Regex, but even after refresh of the filter QRPGLESRC still shows all 10,000+ source members.

**Reply by sebjulliand** [reply]
`F*,J*` is not a regex, it's a regular filter. You can try it out too.

However, I tried it quickly here and it works.

Unfiltered:
![image](https://github.com/user-attachments/assets/c4f6e70a-7906-4b9e-8a91-2da1d6bd615d)

Filtered (showing only members starting with C and F:
![image](https://github.com/user-attachments/assets/0b16fd9d-4f7d-403d-9a8a-9739f9368d6a)
![image](https://github.com/user-attachments/assets/a45bffa3-1618-4167-a02d-43a0c91956a9)

After you refresh your filter, do you see anything weird like an error or something in the Code for i output?

**Comment by thomprl** [comment]
That doesn't seem to work for me either.  It lists every source.  

![image](https://github.com/user-attachments/assets/4467df6e-7a7e-4b15-a3bc-fdb6b3fd0006)

![image](https://github.com/user-attachments/assets/204213a0-cfec-4de0-9f17-3974e7dab6f1)

**Reply by sebjulliand** [reply]
What if you try the `Simple` filtering type and `F*,C*` as the Members filter?
![image](https://github.com/user-attachments/assets/47e78396-d093-4c65-87ad-bac678a3e7c3)

**Reply by thomprl** [reply]
Simple does work for me in that case.  I really don't have a use case to do this, was just trying to verify it worked as you stated.

**Reply by sebjulliand** [reply]
And I'm grateful for that! Now I'll need to understand why the Regex version won't work for you and Greg...
For the sake of it, could you try this Regex filter? `^[FC]`
![image](https://github.com/user-attachments/assets/40bc3c79-5b74-4af3-8ade-317f64db4947)


It should show only members starting with C and F.

**Reply by thomprl** [reply]
So it appears ^[FC], ^(F|C) both work as a regex.  But not (F|J).*  So maybe it's not a bug/problem it's just we didn't know how to enter it correctly.  But probably need to let Greg give it a try.

**Comment by thomprl** [comment]
Also ^(F|C) seems to work as a regex.  But not (F|J).*.

**Reply by Gregory1701** [reply]
@thomprl You going to TechXchange in Vegas this October? If so, let's meet up and have a beer!

**Reply by thomprl** [reply]
Unfortunately I don't get to go to the conferences.  Since I don't know a lot about coding for VSCode/TypeScript I try to help where I can.

**Comment by Gregory1701** [comment]
(Sorry for the slow response, been a busy week) Thanks @thomprl and @sebjulliand for talking through this. I changed my filter back to simple and F.*,J.* , which is what I originally had when I first created this post and it filtered correctly! In our library, and our naming conventions, this cuts my QRPGLESRC in this filter from 10,293 results down to about 100; much nicer! I am not sure why this same filter didn't work before, but does now. The only thing I do notice (not sure it's a big deal) is that with this filter, when I hover over the QRPGLESRC, it still reports the entire set of 10,293 source members and not the current filter subset.
![image](https://github.com/user-attachments/assets/9a8ee64d-8664-4f0a-8579-494d16424dea)

(Okay, I'm willing to admit that I can't find my up-carrot key on my keyboard lol) Trying ^(F|J) works, but I need it to be 'F.' or 'J.', the dot is important. ^(F.|J.) doesn't work. Regex has always been a 'black art' to me. But now, as I said before, for whatever reason, the simple filter is working for me.

But, I still get this error when I try to search QRPGLESRC within this filter:
![image](https://github.com/user-attachments/assets/2bb2192d-8bdc-4d18-acea-1dc5d58dd48f)


Thanks again for helping me through this issue!

**Reply by sebjulliand** [reply]
I'll look into the search bug then.
In the meantime, if you want to keep a dot in your regex, you can simply escape it: `^(F|J)\.` <- String starts with `F` or `J`, then a `.`

**Reply by sebjulliand** [reply]
> The only thing I do notice (not sure it's a big deal) is that with this filter, when I hover over the QRPGLESRC, it still reports the entire set of 10,293 source members and not the current filter subset.

That's because the total number of members is retrieved from the file's metadata; it doesn't take the filter into account (the tooltip can be displayed even when the physical file is collapsed.). Showing the actual number of members with regard to the filter would mean we'd have to count the members to display the tooltip too...that might be a bit expansive just for this 😅

**Comment by sebjulliand** [comment]
@Gregory1701 a fix is underway 😉 https://github.com/codefori/vscode-ibmi/pull/2213

**Reply by sebjulliand** [reply]
@Gregory1701 the fix is now available in the latest pre-release (`v2.12.2 (pre-release)`)
